apis/v1/shared_types.go

```diff
@@ -855,3 +855,15 @@ const (
 	// Support: Extended
 	PermanentCookieLifetimeType CookieLifetimeType = "Permanent"
 )
+
+type Fraction struct {
+        // +optional
+        // +kubebuilder:default=100
+        // +kubebuilder:validation:Minimum=0
+        Numerator int32 `json:"numerator"`
```

Comment:
Recommend requiring this (the struct it's included in is already optional) + also removing default for same reason.
```suggestion
        // +kubebuilder:validation:Minimum=0
        Numerator int32 `json:"numerator"`
```